### PR TITLE
fix(recording): Respect the selected recording service.

### PR DESCRIPTION
**Pull request by hristoterezov**

### CODE REVIEW

File: react/features/recording/components/Recording/AbstractStartRecordingDialog.js [hunk 1]
```diff
@@ -107,9 +113,29 @@ class AbstractStartRecordingDialog extends Component<Props, State> {
...
             this._onTokenUpdated();
         }
+
+        // FIXME: Fix the eslint warning. Potentially check if we need to handle
+        // changes of _fileRecordingsServiceEnabled and
+        // _areIntegrationsEnabled()
+        if (_fileRecordingsServiceEnabled || !this._areIntegrationsEnabled()) {
+            // eslint-disable-next-line react/no-did-mount-set-state
+            this.setState({
+                selectedRecordingService: RECORDING_TYPES.JITSI_REC_SERVICE
```
Comment (virtuacoplenny) [inline react/features/recording/components/Recording/AbstractStartRecordingDialog.js]: You could do this check in the constructor set the state then, avoiding the eslint disabling.

File: react/features/recording/components/Recording/AbstractStartRecordingDialog.js [hunk 2]
```diff
@@ -124,6 +150,32 @@ class AbstractStartRecordingDialog extends Component<Props, State> {
         }
     }
 
+    _areIntegrationsEnabled: () => void;
```
Comment (virtuacoplenny) [inline react/features/recording/components/Recording/AbstractStartRecordingDialog.js]: () => boolean

File: react/features/recording/components/Recording/AbstractStartRecordingDialog.js (hunk 1 above)
Comment (hristoterezov) [reply react/features/recording/components/Recording/AbstractStartRecordingDialog.js]: Good catch! It's fixed!

File: react/features/recording/components/Recording/AbstractStartRecordingDialog.js (hunk 2 above)
Comment (hristoterezov) [reply react/features/recording/components/Recording/AbstractStartRecordingDialog.js]: Fixed!

**Pull request merged**